dbt_subprojects/daily_spellbook/models/_metrics/transactions/metrics_transactions_daily.sql

```diff
@@ -9,39 +9,58 @@
         )
 }}
 
-
-
-with raw_tx as (
-    select * from (
-        select
-            blockchain
-            , cast(date_trunc('day', block_time) as date) as block_date
-            , hash as tx_hash
-        from
-            {{ source('evms', 'transactions') }}
-        where
-            1 = 1
-            {% if is_incremental() %}
-            and {{ incremental_predicate('block_time') }}
-            {% endif %}
-        group by
-            blockchain
-            , cast(date_trunc('day', block_time) as date)
-            , hash
-    ) union all
-    select * from (
-        select
-            'tron' as blockchain
-            , cast(date_trunc('day', block_time) as date) as block_date
-            , hash as tx_hash
-        from
-            {{ source('tron', 'transactions') }}
-        where
-            1 = 1
-            {% if is_incremental() %}
-            and {{ incremental_predicate('block_time') }}
-            {% endif %}
-    )
+with evm as (
+    select
+        blockchain
+        , cast(date_trunc('day', block_time) as date) as block_date
+        , hash as tx_hash
+    from
+        {{ source('evms', 'transactions') }}
+    where
+        1 = 1
+        {% if is_incremental() or true %}
+        and {{ incremental_predicate('block_time') }}
+        {% endif %}
+), tron as (
+    select
+        'tron' as blockchain
+        , cast(date_trunc('day', block_time) as date) as block_date
+        , hash as tx_hash
+    from
+        {{ source('tron', 'transactions') }}
+    where
+        1 = 1
+        {% if is_incremental() or true %}
+        and {{ incremental_predicate('block_time') }}
+        {% endif %}
+), solana as (
+    select
+        'solana' as blockchain
+        , block_date
+        , id as tx_hash
+    from
+        {{ source('solana', 'transactions') }}
+    where
+        1 = 1
+        and block_date is not null --200m+ rows of nulls
```

Comment:
this is an odd finding here
either of you know why there are so many rows with null `block_date`?

Reply:
no clue, sadly :confused:

Reply:
solscan seems equally confused: https://solscan.io/block/31748087 

probably some client issue, looks like that block slot range that doesn't have dates attached to it was 4 years ago, a few of them have timestamps and are dated around that time: https://solscan.io/block/38877915 

Can safely ignore from my POV

Reply:
fwiw, `solana.blocks` also has the nulls (as expected on your findings here). i will leave the filter then, thanks both